framework/docs/source/how-to-use-built-in-mods.rst

```diff
@@ -21,80 +23,140 @@ is as follows:
 
 .. code-block:: python
 
-    ClientApp = Callable[[Message, Context], Message]
-    Mod = Callable[[Message, Context, ClientApp], Message]
+    ClientAppCallable = Callable[[Message, Context], Message]
+    Mod = Callable[[Message, Context, ClientAppCallable], Message]
 
 A typical mod function might look something like this:
 
 .. code-block:: python
 
-    def example_mod(msg: Message, ctx: Context, nxt: ClientApp) -> Message:
+    from flwr.client.typing import ClientAppCallable
+    from flwr.common import Context, Message
+
+
+    def example_mod(msg: Message, ctx: Context, call_next: ClientAppCallable) -> Message:
         # Do something with incoming Message (or Context)
-        # before passing to the inner ``ClientApp``
-        msg = nxt(msg, ctx)
+        # before passing it to the next layer in the chain.
+        # This could be another Mod or, if this is the last Mod, the ClientApp itself.
+        msg = call_next(msg, ctx)
         # Do something with outgoing Message (or Context)
         # before returning
         return msg
 
 Using Mods
 ----------
 
-To use mods in your ``ClientApp``, you can follow these steps:
+Mods can be registered in two ways: **Application-wide mods** and **Function-specific
+mods**.
 
-1. Import the required mods
-~~~~~~~~~~~~~~~~~~~~~~~~~~~
+1. **Application-wide mods**: These mods apply to all functions within the
+   ``ClientApp``.
+2. **Function-specific mods**: These mods apply only to a specific function (e.g, the
+   function decorated by ``@app.train()``)
 
-First, import the built-in mod you intend to use:
+1. Registering Application-wide Mods
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. code-block:: python
+To use application-wide mods in your ``ClientApp``, follow these steps:
 
-    import flwr as fl
-    from flwr.client.mod import example_mod_1, example_mod_2
-
-2. Define your client function
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-Define your client function (``client_fn``) that will be wrapped by the mod(s):
+Import the required mods
+++++++++++++++++++++++++
 
 .. code-block:: python
 
-    def client_fn(cid):
-        # Your client code goes here.
-        return  # your client
+    import flwr as fl
+    from flwr.client.mod import example_mod_1, example_mod_2
```

Comment:
Does this actually work? Or should we use something here that is actually available?